src/range.rs

```diff
@@ -378,10 +429,60 @@ fn group_adjacent_locations(
 impl<V: Ord + Clone> Range<V> {
     /// Computes the union of this `Range` and another.
     pub fn union(&self, other: &Self) -> Self {
```

Comment:
Could we keep a private canonical implementation? and maybe check the optimized one against the canonical one when not compiling in release? Or maybe just property tests are enough?

Reply:
Yes, we have both: https://github.com/astral-sh/pubgrub/blob/6715dc75d24a5f4f68513f53db76221d1f6bdc91/src/version_set.rs#L52-L59 and https://github.com/astral-sh/pubgrub/blob/74f61fb9cef608d8475f1e00c8d0158ada12e009/src/range.rs#L954-L962

Reply:
Yes, the property test is great. What I was wondering is if we should have both impls in the type, something like
```rs
// inside impl Range
pub fn union(&self, other: &Self) -> Self {
  // optimized one
  ...
  // when not in release, check against union_canonical
  ...
}
fn union_canonical(&self, other: &Self) -> Self {
  // canonical impl
}
```
And in prop tests, we can use union_canonical directly for comparisons.
But that's probably not pragmatic at all to have the check on each optimized function call when not compiling in release, and the prop test is largely enough.
Yeah I think so. Forget about this one.